### ISSUE TITLE
Campaign subject not used

### ISSUE THREAD

**Issue by Propaganistas** [spatie/mailcoach-support#178]
I'm on Mailcoach 2.11.3

I've configured a campaign with the following settings and I'm using Mailgun:
![Capture](https://user-images.githubusercontent.com/6680176/81298300-e0c78680-9074-11ea-86f2-596ed5211fae.PNG)

Luckily I sent the campaign with a test list first, because apparently the set subject isn't retained:

![Capture2](https://user-images.githubusercontent.com/6680176/81298359-f76ddd80-9074-11ea-88ff-b740d75a6db0.PNG)

Seems to happen with both immediate and scheduled delivery.
Test mails do use the set subject...

**Comment by Propaganistas** [comment]
Looking in the database, it seems the `mailcoach_campaigns.subject` column gets emptied, when sending a campaign...

**EDIT:** I just created a new campaign and launched sending while refreshing the database. The subject is indeed being emptied while the campaign is in "sending" status!

**Comment by freekmurze** [comment]
Verifying...

**Comment by freekmurze** [comment]
I'm trying to recreate the project using a test. Seems like this one is passing: https://github.com/spatie/laravel-mailcoach/blob/faf788b7ab96724a29a0e8bcca94222312281de9/tests/Jobs/SendCampaignJobTest.php#L88-L101

@Propaganistas  Can you provide a failing test for me to fix. Will test in a real world app now...

**Comment by freekmurze** [comment]
I'm using `v2.11.4` of mailcoach on freek.dev. There, using a custom subject seems to be working fine.

<img width="456" alt="Screenshot 2020-05-07 at 16 03 47" src="https://user-images.githubusercontent.com/483853/81303843-8b8f7300-907c-11ea-9c27-375ccf987756.png">

**Comment by freekmurze** [comment]
O, noticed that you are using the test mail, not the actual campaign...

**Comment by freekmurze** [comment]
Seems test mail subject are working fine as well.
<img width="799" alt="Screenshot 2020-05-07 at 16 08 23" src="https://user-images.githubusercontent.com/483853/81304208-0e183280-907d-11ea-9e0b-48e5e4aa1124.png">

<img width="454" alt="Screenshot 2020-05-07 at 16 08 26" src="https://user-images.githubusercontent.com/483853/81304227-12445000-907d-11ea-97e7-f33369149692.png">

**Comment by freekmurze** [comment]
This test proves that the subject should be ok, even for a test mail: https://github.com/spatie/laravel-mailcoach/blob/305d23ae09a87786fd4f94b26d70c3f5b1db946f/tests/Jobs/SendTestMailJobTest.php#L31

I appreciate any tips on how to reproduce the problem you are seeing.

**Comment by Propaganistas** [comment]
I'm on a completely fresh install of Mailcoach 2.11.3. No fancy settings... Let's set up a call? I'll email you.

**Comment by freekmurze** [comment]
Unfortunately, I don't have time for a call atm. Could you try upgrading to the latest version?

**Comment by Propaganistas** [comment]
Okay got it. I did configure a custom replacer that merely substitutes a placeholder for the current year. Nothing too fancy, almost copy-paste from the docs.

But here's the catch. The docs are leading to this "bug":

https://mailcoach.app/docs/v2/package/advanced-usage/creating-custom-placeholders#creating-a-replacer

```php
public function replace(string $html, Campaign $campaign): string
{
    $webviewUrl = $campaign->webviewUrl();

    return str_ireplace('::webviewUrl::', $webviewUrl, $campaign->email_html);
}
```

should be 

```php
public function replace(string $html, Campaign $campaign): string
{
    $webviewUrl = $campaign->webviewUrl();

    return str_ireplace('::webviewUrl::', $webviewUrl, $html);
}
```

It seems that performing substitution and returning `$campaign->email_html` again loses the subject somehow.

I guess this is sorted out. Sorry to make all the fuss, but I did everything by the book ;-) You might want to update your documentation.
